source/beamline/PGM/PGMBpmControl.cpp

```diff
@@ -0,0 +1,36 @@
+#include "PGMBpmControl.h"
+#include <qdebug.h>
+PGMBpmControl::PGMBpmControl(const QString &name, const QString &pvName, const int avgValue, const int variance, QObject *parent)
+    : QObject(parent)
+
+{
+    currentValue_ = 0;
+    averageValue_ = avgValue;
+    variance_ = variance;
+
+    bpmPV_ = new AMReadOnlyPVControl(name, pvName, this);
+
+    // Get updated pv value from AMReadOnlyPVControl.
+    connect(bpmPV_, SIGNAL(valueChanged(double)), this, SLOT(setCurrentValue(double)));
+
+}
+
+void PGMBpmControl::isValid(){
```

Comment:
This function doesn't really make sense.  `isValid()` should return a bool and if you want the valid to change state you should use a more traditional signal/slot mechanism to update it.  Something like:

```c++
// In constructor
connect(bpmPV_, SIGNAL(valueChanged(double)), this, SLOT(updateValidity(double)));


// Elsewhere in the class
bool PGMBPMControl::updateValidity(double newValue)
{
   int upperBound = averageValue_ + variance_;
   int lowerBound = averageValue_ - variance_;
   bool newState = (newValue < upperBound || newValue > lowerBound);

   if (newState != state_){
       
      state_ = newState;
      emit stateValidityChanged(state_);
   }
}
```